c++/src/kj/compat/http.c++

```diff
@@ -6312,9 +6396,14 @@ public:
     KJ_REQUIRE(!headers.isWebSocket(), "WebSocket upgrade headers are not permitted in a connect.");
     return client.connect(host, headers)
         .then([&tunnel](HttpClient::ConnectResponse&& response) -> kj::Promise<void> {
-      KJ_SWITCH_ONEOF(response.connectionOrBody) {
-        KJ_CASE_ONEOF(io, kj::Own<kj::AsyncIoStream>) {
-          auto io2 = tunnel.accept(response.statusCode, response.statusText, *response.headers);
+      // TODO(someday): The adapter here currently does not support pipelining. We have to
```

Comment:
This seems like it will defeat the purpose in `workerd`, since all subrequest channels there are represented as `WorkerInterface` which inherits `HttpService`, not `HttpClient`. So you will always end up using the adapter...

Reply:
Oof, yeah, good point.